File: microservices/device-service/src/models/device.ts
import { model, Schema, Model, Document } from "mongoose";

interface IDevice {
    name: string;
    state: [
        {
            updated: Date;
            connection: string;
            value: string;
        }
    ];
}

const DeviceSchema: Schema = new Schema({
    name: { type: String },
    state: [
        {
            updated: { type: Date },
            connection: { type: String },
            value: { type: String },
        },
    ],
});

const Device: Model<IDevice> = model("User", DeviceSchema);

export = Device;

File: app/lib/models/device.dart
class State {
  final String date;
  final String connection;
  final String value;
  const State(this.date, this.connection, this.value);
}

class Device {
  final String id;
  final String name;
  final List<State> state;

  const Device(this.id, this.name, this.state);
}
